src/data/trainers.h

```diff
@@ -8,6 +8,8 @@
 
 #line 1 "src/data/trainers.party"
 
+[DIFFICULTY_NORMAL] =
+{
 #line 76
     [TRAINER_NONE] =
     {
```

Comment:
Instead of:
```c
[DIFFICULTY_a] = {
  [TRAINER_a] = { ... },
  [TRAINER_b] = { ... },
},
[DIFFICULTY_b] = {
  [TRAINER_a] = { ... },
  [TRAINER_b] = { ... },
},
```
Could you make `trainerproc` produce:
```c
[DIFFICULTY_a][TRAINER_a] = { ... },
[DIFFICULTY_b][TRAINER_a] = { ... },
[DIFFICULTY_a][TRAINER_b] = { ... },
[DIFFICULTY_b][TRAINER_b] = { ... },
```
Or
```c

[DIFFICULTY_a][TRAINER_a] = { ... },
[DIFFICULTY_a][TRAINER_b] = { ... },
[DIFFICULTY_b][TRAINER_a] = { ... },
[DIFFICULTY_b][TRAINER_b] = { ... },
```
(Whichever is easier, the exact order doesn't matter with this initialization syntax.)

I think that should preserve the ability to split trainers over multiple `.party` files... and maybe simplifies `trainerproc`, because I suppose it allows us to output trainers in whatever order they're parsed rather than having to batch up the difficulties and output them all at the end.

Reply:
I will work on this after @SBird1337's [pull request](https://discord.com/channels/419213663107416084/1207720924414087168/1284059815533875242), since I have no idea what changes that's going to make.

Reply:
[`2d495cf` (#5337)](https://github.com/rh-hideout/pokeemerald-expansion/pull/5337/commits/2d495cfb4b5bd016833f02402076f96a4ace109e)